File: Creational/Builder/Java/src/main/java/com/fastfood/MealDirector.java
package main.java.com.fastfood;

public class MealDirector {
    private MealBuilder mealBuilder;

    public MealDirector(MealBuilder mealBuilder) {
        this.mealBuilder = mealBuilder;
    }

    public Meal constructMeal() {
        mealBuilder.buildMainItem();
        mealBuilder.buildSideItem();
        mealBuilder.buildDrink();
        mealBuilder.buildToy();
        return mealBuilder.getMeal();
    }
}

File: Creational/Builder/C#/MealDirector.cs
namespace FastFoodBuilder {
    public class MealDirector {
        private readonly IMealBuilder _mealBuilder;

        public MealDirector(IMealBuilder mealBuilder) {
            _mealBuilder = mealBuilder;
        }

        public Meal ConstructMeal() {
            _mealBuilder.BuildMainItem();
            _mealBuilder.BuildSideItem();
            _mealBuilder.BuildDrink();
            _mealBuilder.BuildToy();
            return _mealBuilder.GetMeal();
        }
    }
}
